### PR TITLE
Implement assert with backtraces

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/common/config.h.in [hunk 1]
```diff
@@ -25,5 +25,6 @@
 #cmakedefine AWS_HAVE_GCC_OVERFLOW_MATH_EXTENSIONS
 #cmakedefine AWS_HAVE_GCC_INLINE_ASM
 #cmakedefine AWS_HAVE_MSVC_MULX
+#cmakedefine AWS_HAVE_EXECINFO
 
 #endif
```
Comment (JonathanHenson) [inline include/aws/common/config.h.in]: do we need a newline here?

Comment (justinboswell) [reply include/aws/common/config.h.in]: It was there before, so I left it. I don't think so, looking at the output. Does it offend you so? :)

Comment (justinboswell) [reply include/aws/common/config.h.in]: Oh I see, you meant at the end. I've added one.

**Comment by danielsn** [comment]
The easiest way to avoid the CBMC errors would be to the new asserts in ```#ifndef CBMC```, and use standard asserts in CBMC mode.

**Pull request merged**